src/initialize-custom-elements.ts

```diff
@@ -20,13 +20,14 @@ function initializeCustomElement(app: Application, name: string): void {
         parent.insertBefore(placeholder, this);
         parent.removeChild(this);
 
-        app.renderComponent(name, parent, placeholder).then(() => {
-          let customElement = this as Element;
-          let glimmerElement = placeholder.previousElementSibling;
+        app.renderComponent(name, parent, placeholder);
+        waitForRenderSync(app);
```

Comment:
I don't think we want this to be sync.  Maybe we could do a simplified version of the did render helper?

```js
whenRendered(app, () => {
  let customElement = this as Element;
  let glimmerElement = placeholder.previousElementSibling;

  placeholder.remove();
  assignAttributes(customElement, glimmerElement);
});
```

```js
function whenRendered(app, callback) {
  if (app['_rendering']) {
    setTimeout(whenRendered, 10, app, callback);
  } else {
    callback();
  }
}
```